dagstore_test.go

```diff
@@ -211,37 +236,42 @@ func TestConcurrentAcquires(t *testing.T) {
 	res := <-ch
 	require.NoError(t, res.Error)
 
-	t.Run("1", func(t *testing.T) { acquireShard(t, dagst, k, 1) })
-	t.Run("2", func(t *testing.T) { acquireShard(t, dagst, k, 2) })
-	t.Run("4", func(t *testing.T) { acquireShard(t, dagst, k, 4) })
-	t.Run("8", func(t *testing.T) { acquireShard(t, dagst, k, 8) })
-	t.Run("16", func(t *testing.T) { acquireShard(t, dagst, k, 16) })
-	t.Run("32", func(t *testing.T) { acquireShard(t, dagst, k, 32) })
-	t.Run("64", func(t *testing.T) { acquireShard(t, dagst, k, 64) })
-	t.Run("128", func(t *testing.T) { acquireShard(t, dagst, k, 128) })
-	t.Run("256", func(t *testing.T) { acquireShard(t, dagst, k, 256) })
+	t.Run("1", func(t *testing.T) { acquireShardThenRelease(t, dagst, k, 1) })
+	t.Run("2", func(t *testing.T) { acquireShardThenRelease(t, dagst, k, 2) })
+	t.Run("4", func(t *testing.T) { acquireShardThenRelease(t, dagst, k, 4) })
+	t.Run("8", func(t *testing.T) { acquireShardThenRelease(t, dagst, k, 8) })
+	t.Run("16", func(t *testing.T) { acquireShardThenRelease(t, dagst, k, 16) })
+	t.Run("32", func(t *testing.T) { acquireShardThenRelease(t, dagst, k, 32) })
+	t.Run("64", func(t *testing.T) { acquireShardThenRelease(t, dagst, k, 64) })
+	t.Run("128", func(t *testing.T) { acquireShardThenRelease(t, dagst, k, 128) })
+	t.Run("256", func(t *testing.T) { acquireShardThenRelease(t, dagst, k, 256) })
```

Comment:
Let's not overload test helpers. This specialises them too much to the extent that their utility becomes very narrow and diminished.

Can you please have acquire return `[]ShardAccessor`, and create a new test utility `release(...ShardAccessor)` that releases the accessors?

Reply:
Makes a lot of sense.